xwiki-commons-core/xwiki-commons-xml/src/main/java/org/xwiki/xml/internal/html/filter/BodyFilter.java

```diff
@@ -55,11 +55,17 @@
      * "P | %heading; | %list; | %preformatted; | DL | DIV | NOSCRIPT |
      * BLOCKQUOTE | FORM | HR | TABLE | FIELDSET | ADDRESS">
      * }</pre>
+     *
+     * We also use this list for HTML5 where in theory everything is allowed, but we instead only allow flow content
+     * that is not also phrasing content except for {@code <ins>} and {@code <del>} that were already allowed in HTML 4.
```

Comment:
This doesn't have any effect when filtering HTML4? Like allowing ``<article>`` and ``<section>`` inside the body tag when cleaning HTML4.

Reply:
This list is not about allowing additional tags, this list is about which tags should not be wrapped in a `<p>`-tag when they are found as direct children below `<body>`. There are two reasons why I'm confident this is not a problem for cleaning HTML4:

* Most of these tags including `<article>` and `<section>` are not present in `HTML4TagProvider` and thus removed by HTMLCleaner before this filter is even called.  The tags `<menu>` and `<details>` are already in `HTML4TagProvider`, but they definitely shouldn't be wrapped in a paragraph also in HTML4 I think. An exception is `<template>`, where I just noticed that I did not follow the rule to wrap tags that are also phrasing content, but this feels like another exception that is worth making as `<template>`-tags are not rendered and thus having a paragraph around them seems wrong. I have extended the comment to explain this exception.
* Wrapping the listed tags in a paragraph seems wrong, regardless if we are dealing with HTML4 or HTML5.